src/platform_impl/windows/raw_input.rs

```diff
@@ -231,5 +423,265 @@ pub fn get_raw_mouse_button_state(button_flags: USHORT) -> [Option<ElementState>
+        let pre_parsed_data = get_raw_input_pre_parse_info(handle)?;
+        let data_ptr = pre_parsed_data.as_ptr() as PHIDP_PREPARSED_DATA;
+        let mut caps = unsafe { mem::uninitialized() };
+        let status = unsafe { HidP_GetCaps(data_ptr, &mut caps) };
+        if status != HIDP_STATUS_SUCCESS {
+            return None;
+        }
+        let mut button_caps_len = caps.NumberInputButtonCaps;
+        let mut button_caps = Vec::with_capacity(button_caps_len as _);
+        let status = unsafe { HidP_GetButtonCaps(
+            HidP_Input,
+            button_caps.as_mut_ptr(),
+            &mut button_caps_len,
+            data_ptr,
+        ) };
+        if status != HIDP_STATUS_SUCCESS {
+            return None;
+        }
+        unsafe { button_caps.set_len(button_caps_len as _) };
+        let mut button_count = 0;
+        for button_cap in button_caps {
+            let range = unsafe { button_cap.u.Range() };
+            button_count = max(button_count, range.UsageMax);
+        }
+        let button_state = vec![false; button_count as usize];
+        let mut axis_caps_len = caps.NumberInputValueCaps;
+        let mut axis_caps = Vec::with_capacity(axis_caps_len as _);
+        let status = unsafe { HidP_GetValueCaps(
+            HidP_Input,
+            axis_caps.as_mut_ptr(),
+            &mut axis_caps_len,
+            data_ptr,
+        ) };
+        if status != HIDP_STATUS_SUCCESS {
+            return None;
+        }
+        unsafe { axis_caps.set_len(axis_caps_len as _) };
+        let mut axis_state = Vec::with_capacity(axis_caps_len as _);
+        let mut axis_count = 0;
+        for (axis_index, axis_cap) in axis_caps.drain(0..).enumerate() {
+            axis_state.push(Axis {
+                caps: axis_cap,
+                value: 0.0,
+                prev_value: 0.0,
+                axis: None,
+                // @francesca64 when reviewing - where did you get these values? I've commented them
+                // out because the HOTAS controller I've been using to test the raw input backend has
+                // been getting axis hints when it shouldn't, and this seems to be the culprit.
+                // match unsafe { axis_cap.u.Range().UsageMin } {
+                //     0x30 => Some(GamepadAxis::LeftStickX),
+                //     0x31 => Some(GamepadAxis::LeftStickY),
+                //     0x32 => Some(GamepadAxis::RightStickX),
+                //     0x33 => Some(GamepadAxis::LeftTrigger),
+                //     0x34 => Some(GamepadAxis::RightTrigger),
+                //     0x35 => Some(GamepadAxis::RightStickY),
+                //     0x39 => Some(GamepadAxis::HatSwitch),
+                //     0x90 => Some(GamepadAxis::DPadUp),
+                //     0x91 => Some(GamepadAxis::DPadDown),
+                //     0x92 => Some(GamepadAxis::DPadRight),
+                //     0x93 => Some(GamepadAxis::DPadLeft),
+                //     _ => None,
+                // },
```

Comment:
Copied from comment:
@francesca64 when reviewing - where did you get these values? I've commented them out because the HOTAS controller I've been using to test the raw input backend has been getting axis hints when it shouldn't, and this seems to be the culprit.

Reply:
I got them from here: http://www.freebsddiary.org/APC/usb_hid_usages.php

I'm confident those values aren't reliable, of course. Being compatible with SDL mappings is basically the only way to provide layout info for RAWINPUT/etc.